### PR TITLE
[MIG] mail_activity_team: Migration to 14.0

### PR DESCRIPTION
**Pull request by SodexisTeam**
Depends on https://github.com/OCA/social/pull/661

@stephankeller

### CODE REVIEW

**Comment by CasVissers-360ERP** [comment]
I think there is a missing parameter in this module, on base.automation there is no option to set an activity team if state = next_activity. This results in "The assigned user is not member of the team %s " because an incorrect team might be set on the activity.

Demo use case:
1. Create automated action with type "Next activity" for user A
2. Create a ticket with no user assigned
3. If multiple team are in the system the wrong team might be set

**Comment by atchuthan** [comment]
@CasVissers-360ERP We couldn't reproduce this issue.  Could you provide us a video describing the problem such that we can resolve it?

File: mail_activity_team/models/mail_activity.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        # SUPERUSER is inactive and then even if you add it
+        # to member_ids it's not taken account
+        # To not be blocked we must add it to constraint condition
+        for _activity in self.filtered(
+            lambda a: a.user_id.id != SUPERUSER_ID
+            and a.team_id
+            and a.user_id
+            and a.user_id not in a.team_id.member_ids
+        ):
+            raise ValidationError(_("The assigned user is not member of the team."))
```
Comment (CasVissers-360ERP) [inline mail_activity_team/models/mail_activity.py]: Might be better to add user to error?
`raise ValidationError(_("The assigned user %s is not member of the team %s.") % (self.user_id.name, self.team_id.name))`

Comment (atchuthan) [reply mail_activity_team/models/mail_activity.py]: added

**Review by chienandalu — CHANGES_REQUESTED** [review]
Please include this commit https://github.com/OCA/social/commit/6a96684e50a12b067d03eb7da381a5b812e1fa04

It should fix the issue commented by @CasVissers-360ERP

**Comment by pedrobaeza** [comment]
Please include #697 and fix CI.

**Comment by MiquelRForgeFlow** [comment]
And also include https://github.com/OCA/social/pull/729.

**Comment by atchuthan** [comment]
> Please include #697 and fix CI.

added the fix with commit https://github.com/OCA/social/pull/662/commits/67792ea5f6b4836408e9d0f6bd03f5d296cf3342

**Comment by atchuthan** [comment]
> And also include #729.

added the fix with commit https://github.com/OCA/social/pull/662/commits/32e9a4145b91329626ab2cf8565b038cafb7c13d

**Comment by AaronHForgeFlow** [comment]
Hi @atchuthan can you attend travis here? Are you still working on this? Thank you.

**Comment by atchuthan** [comment]
@AaronHForgeFlow Updated latest code from 14.0 branch

**Comment by MiquelRForgeFlow** [comment]
@atchuthan please write:

`/ocabot rebase`

**Comment by AaronHForgeFlow** [comment]
What is the status of this PR at this moment?

**Comment by atchuthan** [comment]
> What is the status of this PR at this moment?

It is working in one of our customers successfully.  We will try to complete the test file in few days.

**Review by JoanMForgeFlow — APPROVED** [review]
LGTM!

File: mail_activity_team/tests/test_mail_activity_team.py [hunk 2]
```diff
@@ -139,15 +139,6 @@ def test_missing_activities(self):
         self.assertEqual(self.team1.count_missing_activities, 0)
         self.assertEqual(self.act1.team_id, self.team1)
 
-    def test_team_onchanges(self):
```
Comment (AaronHForgeFlow) [inline mail_activity_team/tests/test_mail_activity_team.py]: why is this removed?

Comment (atchuthan) [reply mail_activity_team/tests/test_mail_activity_team.py]: When a new member is added, the onchange clears the value in leader(i.e. set to null), so we removed it in migration.

**Review by JoanMForgeFlow — APPROVED** [review]

**Review by atchuthan — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review LGTM

**Comment by AaronHForgeFlow** [comment]
@chienandalu could you update your review? thanks

**Review by chienandalu — APPROVED** [review]

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Can you drop empty commits? (the merge ones) Thanks!

**Comment by atchuthan** [comment]
> Can you drop empty commits? (the merge ones) Thanks!

Removed the commits and pushed the changes once again
